src/unix.js

```diff
@@ -74,12 +75,13 @@ function escapeArgDash(arg, interpolation, quoted) {
   if (interpolation) {
     result = result
       .replace(/\\/g, "\\\\")
+      .replace(/\n/g, " ")
       .replace(/(^|\s)(~|#)/g, "$1\\$2")
       .replace(/(\*|\?)/g, "\\$1")
       .replace(/(\$|\;|\&|\|)/g, "\\$1")
       .replace(/(\(|\)|\<|\>)/g, "\\$1")
       .replace(/("|'|`)/g, "\\$1")
-      .replace(/\{(?=(.*?(?:\,|\.).*?)\})/g, "\\{");
+      .replace(/\{(?=([^]*?(?:\,|\.)[^]*?)\})/g, "\\{");
```

Comment:
For Dash, this change isn't actually relevant - see https://github.com/ericcornelissen/shescape/pull/336